### ISSUE TITLE
Spellchecker does not check all files

### ISSUE THREAD

**Issue by cmichi** [paritytech/ink#738]
We have a typo here:
https://github.com/paritytech/ink/blob/59a7f5edc61eb1fefd6b4179c6ec40d16e433d46/crates/env/src/engine/off_chain/mod.rs#L130 (unintialized vs. uninitialized).

This should have been detected by our `cargo-spellcheck` CI stage. If I enter some random gibberish in that doc comment the mistake is also not detected when I run the checker locally:
```
cargo spellcheck check --cfg=.config/cargo_spellcheck.toml --checkers hunspell --code 1
```

**Comment by cmichi** [comment]
@drahnr We have some logic which includes files depending on enabled features and environment (std vs. wasm). Could this be the cause or should `cargo-spellcheck` check everything, independent of if a module is included in the build?

**Comment by drahnr** [comment]
The mod pickup is rather primitive, it just takes whatever `mod $tt` it finds, it does not evaluate any `#[cfg(..)]` annotations which is intentional.

```
cargo spellcheck -vvvvv -r ./path/to/file/with/the/mod/decl/which/are/not/pciked/up.rs
```

**Comment by drahnr** [comment]
This is a bug in the module detection code:
Parent mod
```
cfg_if! {
    if #[cfg(all(not(feature = "std"), target_arch = "wasm32"))] {
        mod on_chain;
        pub use self::on_chain::EnvInstance;
    } else if #[cfg(feature = "std")] {
        pub mod off_chain;
        pub use self::off_chain::EnvInstance;
        pub use self::off_chain::{
            AccountError,
            TypedEncodedError,
        };
//...
}
```

which is not parsed correctly by https://github.com/drahnr/cargo-spellcheck/blob/master/src/traverse/mod.rs#L32-L121

**Comment by drahnr** [comment]
Solved as of `v0.8.0-alpha.0`
